sys/net/gnrc/routing/ipv6_auto_subnets/gnrc_ipv6_auto_subnets.c

```diff
@@ -302,8 +304,8 @@ static void _configure_subnets(uint8_t subnets, uint8_t start_idx, gnrc_netif_t
         _remove_old_prefix(downstream, &new_prefix, new_prefix_len, &ext_opts);
 
         /* configure subnet on downstream interface */
-        gnrc_netif_ipv6_add_prefix(downstream, &new_prefix, new_prefix_len,
-                                   valid_ltime, pref_ltime);
+        idx = gnrc_netif_ipv6_add_prefix(downstream, &new_prefix, new_prefix_len,
+                                         valid_ltime, pref_ltime);
 
         /* start advertising subnet */
         gnrc_ipv6_nib_change_rtr_adv_iface(downstream, true);
```

Comment:
In `dhcpv6_client_conf_prefix()` and  `uhcp_handle_prefix()`, the interface is turned into an advertising interface, only if the interface could configure and add an address from the prefix. So should this line also be moved to the success case?
```C
if (idx >= 0) {
    /* start advertising subnet */
    gnrc_ipv6_nib_change_rtr_adv_iface(downstream, true);
    gnrc_rpl_configure_root(downstream, &downstream->ipv6.addrs[idx]);
}
```
Or is there a reason not to do this?